### ISSUE TITLE
Replace UnitConversion sub-modue by using pint

### ISSUE THREAD

**Issue by stadelmanma** [stadelmanma/netl-AP_MAP_FLOW#42]
I have some odd and overly complex logic in play in UnitDecomposition as well as some extra stuff in some of the conversion classes that doesn't need to be there. 

If I go the pint route then I'll need to update my instructions appropriately. If I require people to use `pip install --user -r requirements.txt` then I'll need to split my requirements file into `conda_requirements.txt` and `pip_requirments.txt` since I don't want pip trying to install stuff like numpy on Windows.

**Comment by stadelmanma** [comment]
After thinking about it more and more, I see no reason to not use [pint](https://github.com/hgrecco/pint)  anymore. Yes it adds a dependency but unit conversion is used so little that I don't think it matters and it will significantly reduce the code base. Tweaks required to match existing functionality:

* Return conversion factors by running `ureg('{input_unit}').to('{output_unit}')`
* To SI conversions can be handled by running `ureg('{input_unit}').to_base_units()`
* Add a method to easily define or redefine a 'voxel' unit http://pint.readthedocs.io/en/0.7.2/defining.html#programmatically
* Update model outputs to be in correct case for units i.e. return Pa instead of PA.
* May need to do some tweaking to ensure temperature conversions still work as expected 
    - pint seems to want it's units defined as `degC` or `Celsius` instead of just `C`, etc.
    - `C` -> columb, `F` -> `farad`, `R` -> `molar_gas_constant`, `K` -> `Kelvin` given the context of my model I will be safe to override these manually using a dictionary mapping them to the `degX` counterpart.

**Comment by stadelmanma** [comment]
Rewriting this to use pint will allow me to remove all of the files in the UnitConversion submodule. I will only need to keep two functions, `get_conversion_factor` and `convert_value` so I can move those into `__core__.py` and instantiate my UnitRegistry there.

This will significantly lighten my test suite as well since I had a lot of corner cases to cover with unit conversion.

**Comment by stadelmanma** [comment]
doing something like the following would allow me to use pint for my unit conversions in Fortran. 
```
python3 -c 'import sys, pint; ureg = pint.UnitRegistry(); sys.stdout.write(str(ureg("kin").to("in").magnitude)+"\n")'
```

I'd run this with `EXECUTE_COMMAND_LINE` and catch the stdout then convert to floating point value. I could essentially run small python commands from within Fortran but I'd need to ensure I could catch the exit code to properly handle errors and preferably have stderr still appear on the screen if that doesn't happen by default.

This also opens the door to some other platform independent things like using `os.remove` instead of `rm -rf file`. The first one will work on any OS, the second will work in Cygwin but not native windows cmd.exe

**Comment by stadelmanma** [comment]
Fixed by #68
